packages/wallet-ui/src/utils/utils.ts

```diff
@@ -79,6 +83,19 @@ export const getHumanReadableAmount = (
   return amountStr.substring(0, indexDecimal + firstNonZeroIndex + 3);
 };
 
+export const getSpendableTotalBalance = (asset: Erc20TokenBalance): string => {
+  const amount = asset.spendableAmount ?? asset.amount;
+  const spendableAmount = getHumanReadableAmount(asset, amount.toString());
+
+  if (asset.spendableAmount === asset.amount) {
+    return `${spendableAmount}`;
+  }
+
+  const totalAmount = getHumanReadableAmount(asset, asset.amount.toString());
+
+  return `${spendableAmount} (${totalAmount})`;
+};
```

Comment:
const amount = asset.spendableAmount ?? asset.amount;
it may create a weird case 

if the asset does not have spendableAmount, so it use asset.amount

and `if (asset.spendableAmount === asset.amount) ` will be false

then at the end , it will become `return `${spendableAmount} (${totalAmount})`;`

Reply:
i will say do something like that 
```suggestion
  if (asset.spendableAmount === undefined) {
       throw new Error("Spendable amount can not be undefined")
  }
  
  const spendableAmount = getHumanReadableAmount(asset, asset.spendableAmount);

  if (asset.spendableAmount === asset.amount) {
    return `${spendableAmount}`;
  }

  const totalAmount = getHumanReadableAmount(asset, asset.amount.toString());

  return `${spendableAmount} (${totalAmount})`;
};
```